### PR TITLE
Fix javadocs

### PR DESCRIPTION
**Pull request by dagnir**
Fix `@link` and other misc javadoc errrors.

### CODE REVIEW

File: core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java [hunk 1]
```diff
@@ -53,7 +57,7 @@
  * </ol>
  * An additional {@link #onExecutionFailure} method is provided that is invoked if an execution fails at any point during the
  * lifecycle of a request, including exceptions being thrown from this or other interceptors.
- * </p>
+ * <p></p>
```
Comment (zoewangg) [inline core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java]: do we need these empty tags `<p></p>` here?  I saw them in other files as well

Comment (dagnir) [reply core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java]: Yes, self-closing tags are not allowed.

Comment (zoewangg) [reply core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java]: Gotcha, maybe we can remove the closing tags? from http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html, `</p>` doesn't seem to be needed.

Comment (dagnir) [reply core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java]: Nice, sounds good!

**Pull request merged**